plugins/trace_api_plugin/include/eosio/trace_api/compressed_file.hpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include <ios>
+#include <fc/io/cfile.hpp>
+
+namespace eosio::trace_api {
+
+   class compressed_file_datastream;
+   struct compressed_file_impl;
+   /**
+    * wrapper for read-only access to a compressed file.
+    * compressed files support seeking though the efficiency of random access is lower than
+    * an uncompressed file in trade for smaller on-disk resource usage
+    */
```

Comment:
If the cfile read interface is matched for a code reason (like the unpack requirements below on the stream) then add that to the comment (if it is just because it is better to use the same pattern than to not, then no need for comment)